Fix performance in job details

roggervalf opened the pull request:
In order to prevent a loading time when we have 1000 jobs in our tables, we could just show the logs when we click in job button for standalone page, this PR should solve #220

roggervalf commented:
Please @bradvogel when you have some free time, give some :eyes: on this PR

bradvogel commented on src/server/views/partials/dashboard/jobDetails.hbs:
```diff
@@ -45,15 +43,10 @@
-  <div class="col-sm-3">
-    <h5>Executes At</h5>
-    {{moment (add this.options.delay (getTimestamp this) ) "llll"}}
+    {{moment (add (getDelay this) (getTimestamp this) ) "llll"}}
```
minor thing: it'd be cleaner for getDelay to internalize the math to add the delay. It's a bit confusing to have logic implemented in the template.

bradvogel commented on src/server/views/helpers/handlebars.js:
```diff
@@ -61,8 +61,20 @@ const helpers = {
+
   getTimestamp(job) {
-    return job.timestamp ? job.timestamp : job.options.timestamp;
+    if (job.timestamp) {
```
Small nitpick: This would be a slightly cleaner implementation:

```
  getTimestamp(job) {
    // Bull
    if (job.timestamp) {
      return job.timestamp;
    }

    // Bee
    if (job.options && job.options.timestamp) {
      return job.options.timestamp;
    }
  },
```

bradvogel commented on src/server/views/helpers/handlebars.js:
```diff
@@ -61,8 +61,20 @@ const helpers = {
+      return job.delay;
+    }
+
+    return job.options ? job.options.delay : undefined;
```
See below - I would use getTimestamp() directly in this method in order to keep the business logic out of the template.

roggervalf replied:
sure, I could refactor it in my last commit tyty

bradvogel reviewed (changes requested):
How do I see the logs locally? I tried running `npm run start:bull` in the examples folder, added a job, and let it complete. I opened the direct link to the job (http://localhost:4735/Queue%20Server%201/name_of_my_queue/3) and didn't see "Logs" in the UI

roggervalf commented:
> How do I see the logs locally? I tried running `npm run start:bull` in the examples folder, added a job, and let it complete. I opened the direct link to the job (http://localhost:4735/Queue%20Server%201/name_of_my_queue/3) and didn't see "Logs" in the UI

I added a log example in bull example file, when getting into delayed job should appear it in the standalone page now

beequeueci commented:
:tada: This PR is included in version 3.15.0 :tada:

The release is available on:
- [GitHub release](https://github.com/bee-queue/arena/releases/tag/v3.15.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/bull-arena/v/3.15.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

anuragagarwal561994 commented:
Hello @bradvogel I guess instead of completely disabling this feature we can provide this as an option to the user.
Some of us don't have thousand jobs and more to that if the pagination settings are under control like on one page if we display only 10-20 jobs then it should not make such difference.

bradvogel commented:
Yes, agreed. I'll accept a PR to add an option
